aiida/cmdline/commands/cmd_plugin.py

```diff
@@ -48,11 +49,11 @@ def plugin_list(entry_point_group, entry_point):
             echo.echo_critical(str(exception))
         else:
             try:
-                if issubclass(plugin, Process):
+                if (inspect.isclass(plugin) and issubclass(plugin, Process)) or plugin.is_process_function:
                     print_process_info(plugin)
                 else:
                     echo.echo(str(plugin.get_description()))
-            except (AttributeError, TypeError):
+            except (AttributeError, TypeError) as exception:
                 echo.echo_error('No description available for {}'.format(entry_point))
```

Comment:
>     * I guess `TypeError` here is used to catch the error when plugin is a function, since it is already handled, can be removed.

Note that this can be called also for objects that are neither a process class nor a process function, because this function is used for all entry point groups. But I guess your point still stands, I think for any object, it should ultimately just raise `AttributeError` when we call `plugin.is_process_function`, so I think we can safely just catch that. I will update it.

>     * The `exception` here not used, can we just get rid of it or print it in error message?

Yes, I will remove it.